### ISSUE TITLE
Add automatic code-format to Analysis

### ISSUE THREAD

**Issue by Richargh** [MaibornWolff/codecharta#408]
Add a tool for the Analysis Code that auto-formats code.

**Comment by alexhunziker** [comment]
We have looked at the following options:

- ktlint https://ktlint.github.io/
Is a linter for kotlin. It can be added to the build.gradle as a task. It supports to check the project for conformity with the code style as well as automatic re-formating (by using `gradlew ktlint` and `gradlew  ktlintFormat` respectively). There are not a lot of configuration possibilities, however theoretically new rulesets could be added https://github.com/pinterest/ktlint#creating-a-ruleset. A major downside is that there is not really an option to disable certain rules for the project, instead directives would need to be included in the code.

- .editorconfig
An .editorconfig file could be added to the project. IntelliJ recognizes such a file and applies the specified settings on "Reformate code". The range of supported properties by .editorconfig is relatively small and can be found here: https://editorconfig.org/
We found no possibility to check for these rules during the CI process directly.

- IntelliJ Code Style XML
IntelliJ supports the export of the Code Style Settings as .xml file. This file must then be imported to intelliJ by all developers (that is a downside). However in contrast to the .editorconfig-file, here a wide range of properties can be stored.
